digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/util/AssetDataImporter.scala

```diff
@@ -858,4 +858,201 @@ class AssetDataImporter {
+        println(s"*** Processing municipality: $municipalityCode")
+        val listOfStops = getMTStopsMissingOneAddress(municipalityCode)
+        val roadLinks = vvhClient.fetchByMunicipality(municipalityCode)
+        val finstops=getFinnishStopAddressRSe(municipalityCode)
+        val swedishstops= getSwedishStopAddressRFi(municipalityCode)
+        listOfStops.foreach { stops =>   //stop_1:asset_id, stop_2:link_id
+          roadLinks.foreach { rlinks =>
+            if (rlinks.linkId == stops._2) //stop's link if found from vvh list
+              {
+                var sweAddressAdded=false;
+                finstops.foreach{finstop=> //finstop  _1:asset_id, _2:address,_3:link-id
+                {
+                if (finstop._3==stops._2)
+                  {
+                    val swedishaddress=rlinks.attributes.getOrElse("ROADNAME_SE","none").toString
+                    if (swedishaddress!="none" && swedishaddress!=null && rlinks.attributes.getOrElse("ROADNAME_FI","null").toString.toUpperCase()==finstop._2.toUpperCase())
+                     {
+                       createTextPropertyValue(stops._1, 300096, swedishaddress)
+                       sweAddressAdded=true;
+                     }
+                  }
+                }}
+                  if (!sweAddressAdded)
+                  swedishstops.foreach{swestop=> //swestop  _1:asset_id, _2:address,_3:link-id
+                    if (swestop._3==rlinks.linkId)
+                    {
+                      val finAddress=rlinks.attributes.getOrElse("ROADNAME_FI","none").toString
+                      if (finAddress!="none" && finAddress!=null && rlinks.attributes.getOrElse("ROADNAME_SE","null").toString.toUpperCase()==swestop._2.toUpperCase())
+                      {
+                        createTextPropertyValue(stops._1, 300095, finAddress)
+                      }
+                    }
+
+                  }
+                }
+              }
+          }
+        }
+      }
+    }
+
+    /**
+      * Gets municipalitycodes of stops which have updateable masstransitstops
+      * returns list of int
+      */
+    def getMTStopsMunicipalitycodeBothMissing() =
+    {
+
+      sql"""
+              Select distinct MUNICIPALITY_CODE
+                            From Asset
+                   WHERE
+                   Asset_Type_ID=10
+                    AND
+                    (
+                   (ID NOT IN (SELECT ASSET_ID FROM Text_property_value WHERE PROPERTY_ID=300096 OR PROPERTY_ID=300095 ))
+                    )
+               ORDER BY MUNICIPALITY_CODE DESC""".as[(Int)].list
+    }
+
+  /**
+    * Retrives distinct int list filled with  municipalitycodes of mass transit stops which have only finnish OR swedish address
+    * @return
+    */
+
+  def getMTStopsMunicipalitycodeOneMissing() =
+  {
+
+    sql"""
+                    Select distinct a.MUNICIPALITY_CODE
+                    From Asset a, Text_property_value v, Text_property_value fiv
```

Comment:
I think the alias "v" for the table "Text_property_value" is not used at the rest of the query.